modules/custom-css/custom-css-in-customizer.php

```diff
@@ -0,0 +1,312 @@
+			'default' => '',
+			'transport' => 'postMessage',
+		) ) );
+
+		self::$setting_order[] = 'css';
+
+		$wp_customize->add_setting( 'jetpack_custom_css[preprocessor]', array(
+			'default' => '',
+			'transport' => 'postMessage',
+		) );
+
+		self::$setting_order[] = 'preprocessor';
+
+		$wp_customize->add_setting( 'jetpack_custom_css[replace]', array(
+			'default' => false,
+			'transport' => 'refresh',
+		) );
+
+		self::$setting_order[] = 'replace';
+
+		$wp_customize->add_setting( 'jetpack_custom_css[content_width]', array(
+			'default' => '',
+			'transport' => 'refresh',
+		) );
+
+		self::$setting_order[] = 'content_width';
+
+		$section = new WP_Customize_Section( $wp_customize, 'jetpack_custom_css', array(
+			'title' => __( 'CSS', 'jetpack' ),
+			'priority' => 100,
+		) );
+
+		$section->category = 'cat-design';
+
+		$wp_customize->add_section( $section );
+
+		$wp_customize->add_control( new Jetpack_Custom_CSS_Control( $wp_customize, 'jetpack_custom_css_control', array(
+			'label' => __( 'Custom CSS', 'jetpack' ),
+			'section' => 'jetpack_custom_css',
+			'settings' => 'jetpack_custom_css[css]',
+		) ) );
+
+		$wp_customize->add_control( 'wpcom_custom_css_content_width_control', array(
+			'type' => 'text',
+			'label' => __( 'Media Width', 'jetpack' ),
+			'section' => 'jetpack_custom_css',
+			'settings' => 'jetpack_custom_css[content_width]',
+		) );
+
+
+		$wp_customize->add_control( 'jetpack_css_mode_control', array(
+			'type' => 'checkbox',
+			'label' => __( 'Don\'t use the theme\'s original CSS.', 'jetpack' ),
+			'section' => 'jetpack_custom_css',
+			'settings' => 'jetpack_custom_css[replace]',
+		) );
+
+		do_action( 'jetpack_custom_css_customizer_controls', $wp_customize );
+
+		$preprocessors = apply_filters( 'jetpack_custom_css_preprocessors', array() );
+
+		if ( ! empty( $preprocessors ) ) {
+			$preprocessor_choices = array(
+				'' => __( 'None', 'jetpack' ),
+			);
+
+			foreach ( $preprocessors as $preprocessor_key => $processor )
+				$preprocessor_choices[$preprocessor_key] = $processor['name'];
+
+			$wp_customize->add_control( 'jetpack_css_preprocessors_control', array(
+				'type' => 'select',
+				'choices' => $preprocessor_choices,
+				'label' => __( 'Preprocessor', 'jetpack' ),
+				'section' => 'jetpack_custom_css',
+				'settings' => 'jetpack_custom_css[preprocessor]',
+			) );
+		}
+	}
+
+	/**
+	 * JS needed for the admin portion of the Customizer.
+	 */
+	public static function customize_controls_enqueue_scripts() {
+		wp_register_style( 'jetpack-css-codemirror', plugins_url( 'custom-css/css/codemirror.css', __FILE__ ), array(), '20120905' );
+		wp_enqueue_style( 'jetpack-css-customizer-control', plugins_url( 'custom-css/css/customizer-control.css', __FILE__ ), array( 'jetpack-css-codemirror' ), '20140728' );
+
+		wp_register_script( 'jetpack-css-codemirror', plugins_url( 'custom-css/js/codemirror.min.js', __FILE__ ), array(), '3.16', true );
```

Comment:
In either case, updating CodeMirror would probably be better done as a separate pull request after shipping this.